games-emulation/dosbox/dosbox-0.74-r1.ebuild

```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+inherit eutils games
+
+DESCRIPTION="DOS emulator"
+HOMEPAGE="http://dosbox.sourceforge.net/"
+SRC_URI="mirror://sourceforge/dosbox/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~ppc ~ppc64 ~sparc ~x86"
+IUSE="alsa debug hardened opengl"
+
+DEPEND="alsa? ( media-libs/alsa-lib )
+	opengl? ( virtual/glu virtual/opengl )
+	debug? ( sys-libs/ncurses:0 )
+	media-libs/libpng:0
+	media-libs/libsdl[joystick,video,X]
+	media-libs/sdl-net
+	media-libs/sdl-sound"
+RDEPEND=${DEPEND}
+
+src_prepare() {
+	epatch "${FILESDIR}"/${P}-gcc46.patch
```

Comment:
Stick those patches into the PATCHES array and then call `epatches "${PATCHES[@]}"`